agent/util-scripts/pbench-tool-meister-start

```diff
@@ -288,28 +289,68 @@ def main(argv):
         full_hostname = os.environ["_pbench_full_hostname"]
     except Exception:
         logger.exception("failed to fetch parameters from the environment")
-        return 1
+        return 2
     else:
         tm_dir = Path(benchmark_run_dir, "tm")
         try:
             tm_dir.mkdir()
             os.chdir(tm_dir)
         except Exception:
             logger.exception("failed to create the local tool meister directory")
-            return 1
+            return 3
     if not full_hostname or not hostname:
         logger.error(
             "ERROR - _pbench_hostname ('%s') and _pbench_full_hostname ('%s')"
             " environment variables are required",
             hostname,
             full_hostname,
         )
-        return 1
-    if os.environ.get("_PBENCH_UNIT_TESTS"):
-        # FIXME: this is an artifact of the unit test environment.
-        hostnames = "localhost"
+        return 4
+
+    # Determine the Tool Meister "hostname" to use for the Redis server to
+    # bind to, in addition to "localhost". That same host name will be used by
+    # the Tool Data Sink to bind to as well.
+    tm_bind_hostname = os.environ.get("PBENCH_TM_BIND_HOSTNAME", full_hostname)
+    hostnames_l = []
+    try:
+        localhost_ip = socket.gethostbyname("localhost")
+    except socket.error:
+        # Interesting networking environment, no IP address for "localhost" ...
+        localhost_ip = None
     else:
-        hostnames = f"localhost {full_hostname}"
+        # Add to the list of host names the Redis server will use.
+        hostnames_l.append("localhost")
+    try:
+        tm_bind_hostname_ip = socket.gethostbyname(tm_bind_hostname)
+    except socket.error:
+        # The given Tool Meister host name does not map to an IP address, so
+        # we can't use it.
+        tm_bind_hostname_ip = None
```

Comment:
The assignment isn't necessary as `tm_bind_hostname_ip` isn't referenced outside the `else`. LGTM is complaining. However, a log might be handy here? In any case, you should fix the LGTM alert.